### ISSUE TITLE
A child ingredient should count towards cocktails that can be made

### ISSUE THREAD

**Issue by augustf** [karlomikus/bar-assistant#95]
When checking the "Ingredient is variety of another ingredient" box, it seems to me that cocktails that can be made with the parent ingredient should show as being able to be made. So for instance, if one were to add a particular variety of white rum, and list it as a variety of white rum, any cocktail calling for white rum should count as being able to be made.

**Comment by karlomikus** [comment]
Hello, thanks for the contributing!

I'd have to think about this. I like the current way since it's explicit. I could see this being some kind of toggle when creating the cocktail, to pull in all the "related" ingredients.

Also before doing something like this, the default installation ingredients have to be double checked and modified to have correct relationships.

**Comment by augustf** [comment]
Yes-it's not clearly a bug - more an opinionated sort of thing. This got confusing for me when I had 5 kinds of white rum (all totally interchangeable), but was told I couldn't make white rum cocktails without adding the generic ingredient. But there may be situations (say, whiskey) where parent and child aren't interchangeable. But I suspect those particular cases are a matter of taste and not all that common. The recipe would still say what exact spirit was called for, but you'd at least be shown a list of cocktails that you can plausibly make.

Another more modest way to resolve this might be to automatically add the parent ingredient to the shelf when a child gets added.

**Comment by karlomikus** [comment]
>  This got confusing for me when I had 5 kinds of white rum (all totally interchangeable), but was told I couldn't make white rum cocktails without adding the generic ingredient.

Is there a reason why you didn't add "White rum" to your shelf?

**Comment by augustf** [comment]
Absolutely - and I suspect this as much about the default cocktails provided as anything else. All of them are generic, and so after inputting a large and comprehensive bar, I'm able to make...3. So it really depends what you think the "cocktails you can make" feature means. If it's "exact cocktails you can make with these exact brands of ingredients," it's going to always be a very, very small number. So my personal sense is that it should be "cocktails you could plausibly make with what you have." But that might not work for a lot of people.

A solution here might be to make better use of the substitutes table, or to make it easier to mark a cocktail's requirement for a spirit to be strict or loose. Certainly, all the IBA cocktails included are meant to be generic, and never call for an overly specific kind of ingredient. So in the default data set at least, the "cocktails you can make feature" isn't doing any work at all.

**Comment by zhdenny** [comment]
I utilize the substitutes feature. In addition I try to use generic ingredients for example "Unaged Jamaican Overproof Rum" instead of "Wray & Nephew Overproof Rum" or "RumFire". And currently if the drink really does require me to narrow it down further to something specific, I call that out in instructions/description.

When it comes to base spirits, I try not to over clutter my ingredients list with name brand stuff. Amaro on the other hand really does need to be called out by exact name (Campari for example).

**Comment by karlomikus** [comment]
I can see this being useful depending on how specific you want to manage your ingredients. I still think you can add "Parent ingredient XY" to your shelf since you actually have a variety of that spirit.

> Certainly, all the IBA cocktails included are meant to be generic, and never call for an overly specific kind of ingredient. So in the default data set at least, the "cocktails you can make feature" isn't doing any work at all.

I disagree with this. I think most users won't care about the ingredient specifics (including me), and just want to see what cocktails can they make. Most recipes I add/import are using the generic ingredients.

The good thing is that this change is simple enough that can be implemented with a optional ENV variable, so I plan to add this as a toggle in next version.

**Comment by augustf** [comment]
Yes-I think it would be useful to offer users the option!

To clarify my earlier comment, when I said the feature wasn't doing much work, what I meant was for users who start with the IBA cocktails, and then enter a bunch of spirits they have on their bar. Unless they also add the parent ingredient, they won't see many cocktails they can make. I'm not sure how many people will use the application like that, but it's what I did when I got started.
